pkg/services/govmomi/vcenter/clone.go

```diff
@@ -118,14 +118,27 @@ func Clone(ctx context.Context, vmCtx *capvcontext.VMContext, bootstrapData []by
 		diskMoveType = linkCloneDiskMoveType
 	}
 
-	folder, err := vmCtx.Session.Finder.FolderOrDefault(ctx, vmCtx.VSphereVM.Spec.Folder)
-	if err != nil {
-		return errors.Wrapf(err, "unable to get folder for %q", ctx)
+	var folder *object.Folder
+
+	folderRef := object.ReferenceFromString(vmCtx.VSphereVM.Spec.Folder)
+	if folderRef != nil {
+		folder = object.NewFolder(vmCtx.Session.Client.Client, *folderRef)
+	} else {
+		folder, err = vmCtx.Session.Finder.FolderOrDefault(ctx, vmCtx.VSphereVM.Spec.Folder)
```

Comment:
Should we also add status field and update the names of folders, resource pool  etc which MOIDs  refers to ?

This should be very helpful someone looking at the object to figure out where in VC these resources are created.

Reply:
I honestly would avoid changing the API right now, but I am happy to do it if we think it really adds more value.

If so, I think the best place would be on vSphereVM which IIRC is considered an "internal API" (like intermediate between VsphereMachine and vSphere calls)

Reply:
like i said on this review summary comment, we should probably update API field go doc comments?
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3229#pullrequestreview-2384048320

Reply:
updated. My concern is not about the API field doc (sorry) but adding new status fields.